contracts/MultiCurrency.sol

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.5.0;
+
+library MultiCurrency {
+    function totalSupply(uint256 currencyId) internal view returns (uint256) {
+        uint256[1] memory input;
+
+        uint256 x = 0 << 32;
+        x += currencyId;
+
+        input[0] = x << 216;
+
+        uint256[1] memory output;
+
+        assembly {
+            if iszero(
+                staticcall(gas, 0x0000000000000000001, input, 0x20, output, 0x20)
```

Comment:
can you make the precompiled address a constant so we can change it in one place later

Reply:
Constant variables are not supported by inline assembly.